### DISCUSSION TITLE
How to configure baseOptions as required?

### DISCUSSION THREAD

**Discussion by anderha** [dotansimha/graphql-code-generator#4887] [General] [answered]
Generated hooks have the field `baseOptions` which are optional. If I call a hook from a react component there is no error shown if i miss to hand the parameters over. Is there any way to configure the code generator to generate the fields `baseOptions` and `variables` as required?

I am using version 2.0.7 of graphql-codegen/typescript-react-apollo with apollo/client version 3.2.2.

My types:
```graphql
type Query {
    todo(id: Long!): Todo!
}

type Todo {
    id: Long
    description: String
    done: Boolean
}
```

My query: 
```graphql
query Todo($id: Long!) {
    todo(id: $id) {
        description,
        done
    }
}
```

Generated function:
```ts
export function useTodoQuery(baseOptions?: Apollo.QueryHookOptions<TodoQuery, TodoVariables>) {
    return Apollo.useQuery<TodoQuery, TodoVariables>(TodoDocument, baseOptions);
}
```

Call the hook:
```ts
// query with required parameter:
const { loading, error, data } = useTodoQuery({ variables: {id: 1}})

// query without required parameter:
const { loading, error, data } = useTodoQuery() // no error, because baseOptions are optional -> but I do want an error if i have to hand parameters over
```

**Comment by dotansimha** [comment, marked as answer]
I think you are right, @anderha . This should have better typings configured, and maybe if there are required variables it should not be optional and make sure developers pass the variables (because all the other configurations are really always optional).
Can you please report this as an issue?

**Reply by anderha** [reply]
Thanks for your answer, I opened an [issue](https://github.com/dotansimha/graphql-code-generator/issues/4964)
